### PR TITLE
Setup titles and command line

### PR DESCRIPTION
**Pull request by gperciva**

### CODE REVIEW

**Comment by shinnok** [comment]
Hey,

I took a sweep at your changes in branch:
https://github.com/Tarsnap/tarsnap-gui/commits/gperciva-setup-titles-and-command-line

Changes and fixes:
* Fixed indentation and other style adjustments. Some of your edits were either using space or tab. The C++ style was inconsistent with the project. To simplify this for you later down the road there's a .clang-format file in the root of the repo. If you install clang-format tool you can enforce the project's style in one blow with a one liner:
`clang-format -i src/widgets/setupdialog.cpp`
So you can write however you want, then apply that command on the files changed and voila.
* I switched the header back to the central design keeping the page titles in place. Let's take a look at both variants and see which one looks best.
* Reworked logic and display for CLI detection.
* Fixed advanced expand button.
* Fixed expanding issues for advanced frame and other layout adjustments.
* Made other small adjustments that I deemed fit, now that I'm looking at the setup wizard again.

I think all these changes brings this PR much closer to executing your general idea. You can pull them from my branch on top of yours and let's proceed from there.

**Comment by gperciva** [comment]
Thanks for the `clang-format` trick!  FWIW, I think this requires clang 3.7 or higher, because `AlignConsecutiveDeclarations` is not present in 3.8.  I tried running it with clang 3.8 (which admittedly was only released 2 days ago), and it gives slightly different results than https://github.com/Tarsnap/tarsnap-gui/commits/gperciva-setup-titles-and-command-line .  What version of `clang` are you using?

**Comment by gperciva** [comment]
I'm a big fan of most of these changes!  The only exception is the header... I was hoping to reduce the vertical space required, so that people on smaller screens (or who require screen magnification) can still use tarsnap-gui comfortably.  I don't think the setup wizard needs such a large logo; the user isn't going to forget what she's doing in the 1-2 minutes it takes to go through the wizard.

I sketched some of my future changes here:
https://github.com/gperciva/tarsnap-gui/tree/setup-titles-and-command-line-FIXME-lastpage
This is not a candidate for merging, since the code (and UI) is messy.  (also, I just noticed that the external link on the "complete!" page isn't working on my system due to an "spaces vs. %20" issue.)

As you see, if we use a smaller header, we can make the setup window significantly smaller.  I would be happy to use an `svg` file for that header, so that if somebody with a high-res screen maximizes the setup window for some reason, they'll have a nicely-displayed logo instead of a block small-png-zoom image.  (I've been improving the metapost for the logo-drawing; once that's approved by Colin, we'll have nicer svgs for the logo)

**Comment by gperciva** [comment]
We can compare the two screens here:
header-large
header-small

**Comment by shinnok** [comment]
>What version of clang are you using?
` clang-format -version
clang-format version 3.8.0 (trunk)
`
I'm actually in favor of reducing vertical space in this case. I'm also a bit distracted by the page titles under the Tarsnap logo text in the center aligned version, so the less confusion the better.

**Comment by shinnok** [comment]
Regarding the done page info text, I removed the HTML rich text since it was to be blamed for all kinds of problems due to font styles and sizes for different platforms and lack of ability not have Qt Designer update it witch each edit of the .ui file. I think you got that on your own, I see a QLabel in your updates. If you enable openExternalLinks property on it and then pass the local file url through QUrl::fromUserInput(url).toLocalFile() should produce the expected platform aware url that would be clickable.

**Comment by gperciva** [comment]
For `clang-format`, I'm using the official 3.8 release:

```
$ ~/Downloads/clang+llvm-3.8.0-x86_64-linux-gnu-ubuntu-14.04/bin/clang-format -version
clang-format version 3.8.0 (tags/RELEASE_380/final)
```

**Comment by gperciva** [comment]
(sorry about the lack of `openExternalLinks` -- I added that after uploading the PR, and I was about to update the PR but then I got distracted by the `clang-format` differences, and forgot that I hadn't uploaded that fix)

**Comment by gperciva** [comment]
PR updated.

**Comment by gperciva** [comment]
The update contains a green "OK" if the binaries are found.  Now that we're always going in to the CLI step, I think it's important to be unambiguous if  there's no problem and the user can keep on clicking.  Of course, I'm open to other ideas on how we can clarify this point.

(but when I was trying the version and just saw "Tarsnap CLI version 1.0.37 detected.", I wasn't certain if I could just hit "proceed" or whether I had to do anything else.)

**Comment by shinnok** [comment]
Well the Proceed button is disabled if something is wrong right? And then there's the info text. I don't think we need green nor red text, there's no harm if the user clicks blindly on Proceed if the button is enabled anyway.

**Comment by gperciva** [comment]
ok, what about using a utf-8 checkmark instead?  PR updated to include ✔

File: src/widgets/setupdialog.cpp [hunk 1]
```diff
@@ -381,8 +381,8 @@ void SetupDialog::setTarsnapVersion(QString versionString)
     }
     else
     {
-        _ui->clientVersionLabel->setText(tr("Tarsnap CLI version ") +
-                                         _tarsnapVersion + tr(" detected."));
+        _ui->clientVersionLabel->setText("Tarsnap CLI version " +
```
Comment (shinnok) [inline src/widgets/setupdialog.cpp]: Make sure you keep text encapsulated in tr() to keep text translatable. I like the checkmark.

**Comment by gperciva** [comment]
fixed `tr()`

**Comment by shinnok** [comment]
Merged. Thanks!

**Pull request merged**